src/Pages/Cart/Cart.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import ShoppingProduct from '../../Components/Cart/ShoppingProduct/ShppingProduct';
+import EmptyCart from '../../Components/Cart/EmptyCart/EmptyCart';
+import './Cart.scss';
+
+class Cart extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      quantity: 1,
+      listData: [],
+      selectedArr: [],
+      deletedArr: [],
+      resultPrice: 0,
+      selectedAll: true,
+    };
+  }
+
+  componentDidMount() {
+    fetch('/data/Cart/CartData.json')
+      .then(res => res.json())
+      .then(data => {
+        this.setState(
+          {
+            listData: data.result,
+            selectedArr: Array(data.result.length).fill(true),
+          },
+          this.calculatePrice
+        );
+      });
+  }
+
+  handleSelectedAll = () => {
+    this.setState(state => ({
+      selectedAll: !state.selectedAll,
+    }));
+  };
+
+  inputAllCheckBox = () => {
+    const checkedItems = this.state.selectedArr.every(list => list);
+    const newChecked = Array(this.state.selectedArr.length).fill(!checkedItems);
+    this.setState({ selectedArr: newChecked });
+  };
+
+  handleSingleBox = id => {
+    const { selectedArr } = this.state;
+    const newCheck = [...selectedArr];
+    newCheck[id] = !newCheck[id];
+    this.setState({ selectedArr: newCheck }, () => this.handleAllSelector());
+  };
+
+  handleAllSelector = () => {
+    const checkedItems = this.state.selectedArr.every(list => list);
+    checkedItems
+      ? this.setState({ selectedAll: true })
+      : this.setState({ selectedAll: false });
+  };
+
+  handleQuantity = event => {
+    const { listData } = this.state;
+    const { value, className } = event.target;
+
+    if (listData[value - 1].quantity === 1 && className === 'quantityMinus') {
+      return;
+    }
+
+    const changedQuantityData = listData.map(product => {
+      return Number(value) !== product.id
+        ? product
+        : {
+            ...product,
+            quantity:
+              className === 'quantityMinus'
+                ? product.quantity - 1
+                : product.quantity + 1,
+          };
+    });
+
+    this.setState({ listData: changedQuantityData }, this.calculatePrice);
+  };
+
+  deleteProduct = event => {
+    const { listData } = this.state;
+    const { value } = event.target;
+
+    const filteredData = listData.filter(product => {
+      return Number(value) !== product.id;
+    });
+
+    this.setState(
+      {
+        listData: filteredData,
+        selectedArr: Array(filteredData.length).fill(false),
+      },
+      this.calculatePrice
+    );
+  };
+
+  calculatePrice = () => {
+    const { listData } = this.state;
+
+    const result = listData.reduce(
+      (pre, curr) => pre + parseInt(curr.quantity * curr.price),
+      0
+    );
+    this.setState({ resultPrice: result });
```

Comment:
이럴 때 저는 더 깔끔하게 보이게 하기 위해 변수명을 key값과 동일하게 많이 합니다.
```suggestion
    const resultPrice = listData.reduce(
      (pre, curr) => pre + parseInt(curr.quantity * curr.price),
      0
    );
    this.setState({ resultPrice });
```

Reply:
오, 처음 알게되었습니다. 감사합니다!